public/js/factories/achievementsResourceFactory.js

```diff
@@ -0,0 +1,51 @@
+netstix.factory('AchievementsResource', ['$http', '$q', function($http, $q) {
+  var achievementsResource = {};
+
+  achievementsResource.postAchievements = function(title, criteria, points, challengeRepo, badgeLink) {
```

Comment:
Looking at $resource, it appears it could definitely help us DRY this out a bit more. Added to waffle board to look at when we have some time to research it.